.github/workflows/test.yml

```diff
@@ -51,6 +62,17 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
           check-latest: true
           cache: true
+      - if: ${{ matrix.goos=='windows' }}
+        uses: actions/checkout@v4.2.1
+        with:
+          repository: containerd/containerd
+          ref: v1.7.23
+          path: containerd
+          fetch-depth: 1
+      - if: ${{ matrix.goos=='windows' }}
+        name: "Set up CNI"
+        working-directory: containerd
+        run: GOPATH=$(go env GOPATH) script/setup/install-cni-windows
```

Comment:
why only windows requires this?

Reply:
Magic trick. :-)

Reply:
Here is my tentative explanation: somewhere, unit tests does go through a code path that depends on `generateNetworkConfig`, which is going to check for a binary on the system for the corresponding plugin-type name (`GetPluginType`).

For windows, it is looking for "nat", which is only provided by the cni plugins.

For linux, it is looking for "bridge"... which is present on most systems and have nothing to do with CNI - hence we get away with running the unit tests.

I do not think the unit tests themselves actually depend on running the binary (obviously...), but yep, that is making the case that we should look into that code.